app/views/play/CampaignView.coffee

```diff
@@ -949,10 +949,14 @@ module.exports = class CampaignView extends RootView
   startLevel: (levelElement, courseID=null, courseInstanceID=null) ->
     @setupManager?.destroy()
     levelSlug = levelElement.data 'level-slug'
+    levelOriginal = levelElement.data('level-original')
     session = @preloadedSession if @preloadedSession?.loaded and @preloadedSession.levelSlug is levelSlug
     if(me.showHeroAndInventoryModalsToStudents())
-      session.set('codeLanguage', @classroom.get('aceConfig').language)
-    @setupManager = new LevelSetupManager supermodel: @supermodel, levelID: levelSlug, levelPath: levelElement.data('level-path'), levelName: levelElement.data('level-name'), hadEverChosenHero: @hadEverChosenHero, parent: @, session: session, courseID: courseID, courseInstanceID: courseInstanceID 
+      codeLanguage = @classroomLevelMap[levelOriginal]?.get('primerLanguage') or @classroom?.get('aceConfig')?.language
+      options = {supermodel: @supermodel, levelID: levelSlug, levelPath: levelElement.data('level-path'), levelName: levelElement.data('level-name'), hadEverChosenHero: @hadEverChosenHero, parent: @, courseID: courseID, courseInstanceID: courseInstanceID, codeLanguage: codeLanguage}
+    else
+      options = {supermodel: @supermodel, levelID: levelSlug, levelPath: levelElement.data('level-path'), levelName: levelElement.data('level-name'), hadEverChosenHero: @hadEverChosenHero, parent: @, session: session}
```

Comment:
Changes like this could use some explaining.  Please consider including in commit descriptions in the future.